sensors/nxt_i2c_sensor_defs.c

```diff
@@ -25,6 +25,36 @@
 #include "ms_ev3_smux.h"
 #include "ms_nxtmmx.h"
 
+/*
+ * convert raw data of LEGO NXT Temperature Sensor to Celsius
+ */
+static int lego_nxt_temp_celsius_scale(void *context,
```

Comment:
Great, I was hoping that was the case. What I was concerned about was the fact that `min_raw` is `u32`. Does this get cast to `u16` on the basis of `data_type`?

Reading the scaling data on the ev3sources typedata page, I'm confused since it states raw min is -14080.0 and raw max is 32767.0. These don't match the data sheet which give 0xc90 (-880) and 0x7ff (2047) respectively. I'm also curious how the 262.4°F Si Max is handled given the integral nature of `si_max` (below I just truncated - should I be working in x10^decimals there?)